descwl_coadd/tests/test_coadd.py

```diff
@@ -0,0 +1,201 @@
+    bands=['i', 'z'],
+    coadd_dim=101,
+    psf_dim=51,
+):
+    buff = 5
+
+    galaxy_catalog = make_galaxy_catalog(
+        rng=rng,
+        gal_type="exp",
+        coadd_dim=coadd_dim,
+        buff=buff,
+        layout="grid",
+        gal_config={'mag': 22},
+    )
+
+    if psf_type == "ps":
+        se_dim = get_se_dim(coadd_dim=coadd_dim)
+        psf = make_ps_psf(rng=rng, dim=se_dim)
+    else:
+        psf = make_fixed_psf(psf_type=psf_type)
+
+    if stars:
+        star_catalog = StarCatalog(
+            rng=rng,
+            coadd_dim=coadd_dim,
+            buff=buff,
+            density=100,
+        )
+    else:
+        star_catalog = None
+
+    return make_sim(
+        rng=rng,
+        galaxy_catalog=galaxy_catalog,
+        star_catalog=star_catalog,
+        coadd_dim=coadd_dim,
+        psf_dim=psf_dim,
+        epochs_per_band=epochs_per_band,
+        g1=0.02,
+        g2=0.00,
+        bands=bands,
+        psf=psf,
+        dither=dither,
+        rotate=dither,
+    )
+
+
+@pytest.mark.parametrize('dither', [False, True])
+@pytest.mark.parametrize('rotate', [False, True])
+def test_coadds_smoke(dither, rotate):
+    rng = np.random.RandomState(55)
+
+    coadd_dim = 101
+    psf_dim = 51
+
+    bands = ['r', 'i', 'z']
+    sim_data = _make_sim(
+        rng=rng, psf_type='gauss', bands=bands,
+        coadd_dim=coadd_dim, psf_dim=psf_dim,
+        dither=dither, rotate=rotate,
+    )
+
+    # coadd each band separately
+    bdata = sim_data['band_data']
+    for band in bands:
+        assert band in bdata
+        exps = bdata[band]
+
+        coadd = make_coadd_obs(
+            exps=exps,
+            coadd_wcs=sim_data['coadd_wcs'],
+            coadd_bbox=sim_data['coadd_bbox'],
+            psf_dims=sim_data['psf_dims'],
+            rng=rng,
+            remove_poisson=False,  # no object poisson noise in sims
+        )
+
+        coadd_dims = (coadd_dim, )*2
+        psf_dims = (psf_dim, )*2
+        assert coadd.image.shape == coadd_dims
+        assert coadd.psf.image.shape == psf_dims
+
+
+def test_coadd_psgauss_smoke():
+    psf_type = 'gauss'
+    rng = np.random.RandomState(8312)
+    data = _make_sim(rng=rng, psf_type=psf_type)
+
+    extent = data['coadd_bbox'].getDimensions()
+    coadd_dims = (extent.getX(), extent.getY())
+    assert data['coadd_dims'] == coadd_dims
+
+    psf_dims = data['psf_dims']
+
+    for band, exps in data['band_data'].items():
+        coadd = make_coadd_obs(
+            exps=exps,
+            coadd_wcs=data['coadd_wcs'],
+            coadd_bbox=data['coadd_bbox'],
+            psf_dims=psf_dims,
+            rng=rng,
+            remove_poisson=False,
+        )
+
+        assert coadd.image.shape == coadd_dims
+        assert coadd.noise.shape == coadd_dims
+        assert coadd.psf.image.shape == psf_dims
+
+
+@pytest.mark.skipif(
+    "CATSIM_DIR" not in os.environ,
+    reason='simulation input data is not present'
+)
+def test_coadd_pspsf_smoke():
```

Comment:
This test like the one before it seems redundant.

Reply:
this one is power spectrum psf

Reply:
But what does that have to do with the coadding? Simply putting that through is no different than the fixed PSFs.

Reply:
The underlying psf object from the sim works differently, wanted to make sure we could use it properly.

I guess its a test of the sim and that we can use the sim

If you object I can remove it

Reply:
this is the only place where CATSIM_DIR is needed, that tips the scales, will remove it